cpp/src/quantiles/group_quantiles.cu

```diff
@@ -57,19 +57,24 @@ struct quantiles_functor {
       [=] __device__ (gdf_size_type i) {
         gdf_size_type segment_size = group_size[i];
 
-        for (gdf_size_type j = 0; j < num_quantiles; j++) {
-          gdf_size_type k = i * num_quantiles + j;
-          result[k] = detail::select_quantile(values + group_id[i], segment_size,
-                                              d_quantiles[j], interpolation);
-        }
+        auto value = values + group_id[i];
+        thrust::transform(thrust::seq, d_quantiles, d_quantiles + num_quantiles,
+                          result + i * num_quantiles,
+                          [=](auto q) { 
+                            return detail::select_quantile(value,
+                                                           segment_size, 
+                                                           q, 
+                                                           interpolation); 
+                          }
```

Comment:
I prefer to put `});` together -- otherwise, you have to choose where to align the `);`, and it's not obvious where it should be indented to.